src/main/java/com/example/weatherwaveapi/weather/service/WeatherService.java

```diff
@@ -0,0 +1,70 @@
+package com.example.weatherwaveapi.weather.service;
+
+import com.example.weatherwaveapi.model.response.weatherforecastresponse.WeatherForecastResponse;
+import com.example.weatherwaveapi.model.response.weatherresponse.WeatherResponse;
+import com.example.weatherwaveapi.util.urlbuilder.OpenWeatherUrlBuilder;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+
+
+import static org.springframework.http.HttpMethod.GET;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class WeatherService {
+    private static final String EXCEPTION_MESSAGE = "HTTP error occurred while processing request. Exception message: {}";
+    private OpenWeatherUrlBuilder urlBuilder;
+    public RestTemplate restTemplate = new RestTemplate();
+
+    @Autowired
+    public WeatherService(OpenWeatherUrlBuilder urlBuilder) {
+        this.urlBuilder = urlBuilder;
+    }
```

Comment:
If you decided to use @Autowired with constructor initializing, let's initialize restTemplate here too